device/tt_silicon_driver.cpp

```diff
@@ -932,8 +932,8 @@ void tt_SiliconDevice::write_device_memory(const void *mem_ptr, uint32_t size_in
 
     std::int32_t tlb_index = 0;
     std::optional<std::tuple<std::uint64_t, std::uint64_t>> tlb_data = std::nullopt;
-    if(tlbs_init) {
-        tlb_index = map_core_to_tlb(tt_xy_pair(target.x, target.y));
+    if(tlbs_init_per_chip[target.chip]) {
+        tlb_index = map_core_to_tlb_per_chip[target.chip](tt_xy_pair(target.x, target.y));
```

Comment:
This is going to impose a performance penalty.

Reply:
I know, but I'm also ready to argue it's a small one. Do you suggest something to be changed here or is this just a general comment?

Reply:
Just a general comment.  The Metal guys did some measurements and found that UMD's write path takes an order of magnitude longer than the actual write to the device, mainly on account of all of the map lookups.  I don't have any numbers handy, but from memory I want to say the write itself is ~30 nanoseconds and the map lookups put this code path into the low hundreds of nanoseconds.  Obviously this depends on the system...

In principle a hash table lookup is fast.  But relative to a 4 byte write across PCIe, they start to add up.  There's a lot of them.

Not suggesting you change anything.  Just pointing it out, as this area of UMD has received some scrutiny from its consumers in the past.

Reply:
Will be addressed when the time comes as part of #68